### DISCUSSION TITLE
How to require current password before being able to updateUser?

### DISCUSSION THREAD

**Discussion by leafblader1** [supabase/supabase#5341] [Questions] [answered]
GoTrue API doesn't require users to confirm their current password before updating the users fields, including the new password. This is pretty insecure, how can I disable this endpoint (and only ask for password changes through email) or require current password?

**Comment by leafblader1** [comment]
byump?

**Comment by GaryAustin1** [comment]
Well the email password uses the same API so if you shut that off then you can't update anyway.

There is work going on:
https://github.com/supabase/gotrue/pull/309
https://github.com/supabase/gotrue/issues/264

**Comment by J0** [comment, marked as answer]
We now require reauthentication(with current password) by default in order to update password as a non admin user
